src/Ardalis.ApiEndpoints/CodeAnalyzers/EndpointHasExtraPublicMethodCodeFixProvider.cs

```diff
@@ -0,0 +1,128 @@
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Ardalis.ApiEndpoints.CodeAnalyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(EndpointHasExtraPublicMethodCodeFixProvider)), Shared]
+    public class EndpointHasExtraPublicMethodCodeFixProvider : CodeFixProvider
```

Comment:
Code Fix Provider isn't built yet.

Reply:
@ardalis was thinking of making the CodeFixProvider quite simple, it would make the extra public method as internal:

```csharp
//bad 
public object ExtraMethod(){}

// good
internal object ExtraMethod(){}
```

Thoughts?

Reply:
Making it internal or private is easiest. Deleting it would be another option but probably more drastic. If it's easy to have several you could offer all 3 options (private/internal/delete).

Reply:
I wouldn't be to afraid to offer deleting the method. Assuming that source control is enabled to get stuff back is a strategy that I see more and more for these kind of things, and I think this is definitely a valid action you might want to do for a violation like this.

The scenario I'm going over in my head is the following: I'm moving towards this pattern, so I copy my controller, and I start "strangling" it towards the endpoint pattern, for that, I'll remove all the non-handle methods; having a code fix for that adds value (to me).

Reply:
Added fix for `internal` and `private`, no `delete` in this iteration:

![image](https://user-images.githubusercontent.com/1190907/88519788-83721000-cfa7-11ea-9a77-5dbe5ff68438.png)